src/crimson/osd/replicated_recovery_backend.cc

```diff
@@ -892,6 +892,86 @@ ReplicatedRecoveryBackend::trim_pushed_data(
   return {intervals_usable, data_usable};
 }
 
+seastar::future<hobject_t>
+ReplicatedRecoveryBackend::prep_push_target(
+  const ObjectRecoveryInfo &recovery_info,
+  bool first,
+  bool complete,
+  bool clear_omap,
+  ObjectStore::Transaction* t,
+  const map<string, bufferlist> &attrs,
+  bufferlist&& omap_header)
+{
+  if (!first) {
+    return seastar::make_ready_future<hobject_t>(
+      get_temp_recovery_object(recovery_info.soid,
+                               recovery_info.version));
+  }
+
+  ghobject_t target_oid;
+  if (complete) {
+    target_oid = ghobject_t(recovery_info.soid);
+    if (!recovery_info.object_exist) {
+      t->remove(coll->get_cid(), target_oid);
+      t->touch(coll->get_cid(), target_oid);
+      bufferlist bv = attrs.at(OI_ATTR);
+      object_info_t oi(bv);
```

Comment:
Is the `bv` temporary is intentional?

Reply:
yes. otherwise we have to change `object_info_t(ceph::buffer::list& bl)` to `object_info_t(const ceph::buffer::list& bl)`

Reply:
yes. otherwise we have to change `object_info_t(ceph::buffer::list& bl)` to `object_info_t(const ceph::buffer::list& bl)`